src/Core/Form/IdentifiableObject/Builder/FormBuilder.php

```diff
@@ -110,32 +135,49 @@ public function getForm(array $data = [], array $options = [])
      */
     public function getFormFor($id, array $data = [], array $options = [])
     {
+        // Merge data in initial data
         $data = array_merge($this->dataProvider->getData($id), $data);
+
+        // Fetch initial options
+        $initialOptions = $options;
+
+        // Hook action<FormName>FormDataProviderData
+        $this->hookDispatcher->dispatchWithParameters(
```

Comment:
Same here the hook should be called after optionsProvider has been used

Reply:
Fixed.